tests/vip-integrations/test-vip-integrations.php

```diff
@@ -3,52 +3,85 @@
-		});
+	// Registration
+
+	public function test_register_integration_as_class_name_loads() {
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration-class', FakeIntegration::class );
 
-		activate( 'fake-integration' );
+		$integrations->activate( 'fake-integration-class' );
 
-		$this->run_integration_actions();
-		$this->assertTrue( FakeIntegration::$is_integrated );
+		$integrations->load_active();
+		$this->assertTrue( FakeIntegration::$is_loaded );
 	}
 
-	public function test_register_integration_as_instantiated_class() {
+	public function test_register_integration_as_instantiated_class_loads() {
 		$fake_integration = new class() extends Integration {
-			public $is_integrated = false;
+			public $is_loaded = false;
 
-			public function integrate( array $config ): void {
-				$this->is_integrated = true;
+			public function load( array $config ): void {
+				$this->is_loaded = true;
 			}
 		};
 
-		add_action( 'register_vip_integrations', function() use ( $fake_integration ) {
-			Integrations::instance()->register( 'fake-integration', $fake_integration );
-		});
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration-instance', $fake_integration );
 
-		activate( 'fake-integration' );
+		$integrations->activate( 'fake-integration-instance' );
 
-		$this->run_integration_actions();
-		$this->assertTrue( $fake_integration->is_integrated );
+		$integrations->load_active();
+		$this->assertTrue( $fake_integration->is_loaded );
 	}
 
-	private function run_integration_actions() {
-		do_action( 'register_vip_integrations' );
-		do_action( 'activate_vip_integrations' );
-		do_action( 'integrate_vip_integrations' );
+	// Non-activation
+
+	public function test_register_integration_without_activation_does_not_load() {
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration', FakeIntegration::class );
+
+		$integrations->load_active();
+		$this->assertFalse( FakeIntegration::$is_loaded );
+	}
+
+	// Errors
+
+	public function test_double_slug_registration_throws() {
+		$this->expectException( InvalidArgumentException::class );
+
+		$an_integration = new class() extends Integration {
+			public function load( array $config ): void {}
+		};
+
+		$integrations = new Integrations();
+		$integrations->register( 'non-unique-slug', FakeIntegration::class );
+		$integrations->register( 'non-unique-slug', $an_integration );
+	}
+
+	public function test_invalid_class_name_throws() {
+		$this->expectException( InvalidArgumentException::class );
+
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration', FakeIntegration::class . '-does-not-exist' );
+	}
+
+	public function test_non_integration_subclass_throws() {
+		$this->expectException( InvalidArgumentException::class );
+
+		$random_class = new class() {};
+
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration', $random_class );
```

Comment:
Could be simplified with a standard class instance instead of defining a new class:

```suggestion
		$integrations = new Integrations();
		$integrations->register( 'fake-integration', new stdClass() );
```